pkg/kubelet/kubelet_test.go

```diff
@@ -2437,6 +2438,155 @@ func TestHandlePodAdditionsInvokesPodAdmitHandlers(t *testing.T) {
+			pod:                        podWithUIDNameNsSpec("2", "pod2", "foo", *cpuMemPodSpec),
+			podResourceAllocationExist: true,
+			expectedPodResourceAllocation: state.PodResourceAllocation{
+				"2": map[string]v1.ResourceList{
+					cpuMemPodSpec.Containers[0].Name: cpuMemPodSpec.Containers[0].Resources.Requests,
+				},
+			},
+		},
+		{
+			name:                       "Only has cpu, resource allocation not exists",
+			pod:                        podWithUIDNameNsSpec("3", "pod3", "foo", *cpuPodSpec),
+			podResourceAllocationExist: false,
+			expectedPodResourceAllocation: state.PodResourceAllocation{
+				"3": map[string]v1.ResourceList{
+					cpuPodSpec.Containers[0].Name: cpuPodSpec.Containers[0].Resources.Requests,
+				},
+			},
+		},
+		{
+			name:                       "Only has cpu, resource allocation exists",
+			pod:                        podWithUIDNameNsSpec("4", "pod4", "foo", *cpuPodSpec),
+			podResourceAllocationExist: true,
+			expectedPodResourceAllocation: state.PodResourceAllocation{
+				"4": map[string]v1.ResourceList{
+					cpuPodSpec.Containers[0].Name: cpuPodSpec.Containers[0].Resources.Requests,
+				},
+			},
+		},
+		{
+			name:                       "Only has memory, resource allocation not exists",
+			pod:                        podWithUIDNameNsSpec("5", "pod5", "foo", *memPodSpec),
+			podResourceAllocationExist: false,
+			expectedPodResourceAllocation: state.PodResourceAllocation{
+				"5": map[string]v1.ResourceList{
+					memPodSpec.Containers[0].Name: memPodSpec.Containers[0].Resources.Requests,
+				},
+			},
+		},
+		{
+			name:                       "Only has memory, resource allocation exists",
+			pod:                        podWithUIDNameNsSpec("6", "pod6", "foo", *memPodSpec),
+			podResourceAllocationExist: true,
+			expectedPodResourceAllocation: state.PodResourceAllocation{
+				"6": map[string]v1.ResourceList{
+					memPodSpec.Containers[0].Name: memPodSpec.Containers[0].Resources.Requests,
+				},
+			},
+		},
+		{
+			name:                       "No CPU and memory, resource allocation not exists",
+			pod:                        podWithUIDNameNsSpec("7", "pod7", "foo", *emptyPodSpec),
+			podResourceAllocationExist: false,
+			expectedPodResourceAllocation: state.PodResourceAllocation{
+				"7": map[string]v1.ResourceList{
+					emptyPodSpec.Containers[0].Name: emptyPodSpec.Containers[0].Resources.Requests,
+				},
+			},
+		},
+		{
+			name:                       "No CPU and memory, resource allocation exists",
+			pod:                        podWithUIDNameNsSpec("8", "pod8", "foo", *emptyPodSpec),
+			podResourceAllocationExist: true,
+			expectedPodResourceAllocation: state.PodResourceAllocation{
+				"8": map[string]v1.ResourceList{
+					emptyPodSpec.Containers[0].Name: emptyPodSpec.Containers[0].Resources.Requests,
+				},
+			},
+		},
+	}
+	for _, tc := range tests {
+		if tc.podResourceAllocationExist {
```

Comment:
Can I ask for one small change? Could you please set the allocation to a value (cpu only, mem only, cpu and mem) that is different from the value in tc.pod and adjust expected accordingly? This will test the cases where `desired != allocated`